adagrams/game.py

```diff
@@ -1,11 +1,106 @@
+import random
+
 def draw_letters():
-    pass
+    LETTER_POOL = {
+    "A": 9,
+    "B": 2,
+    "C": 2,
+    "D": 4,
+    "E": 12,
+    "F": 2,
+    "G": 3,
+    "H": 2,
+    "I": 9,
+    "J": 1,
+    "K": 1,
+    "L": 4,
+    "M": 2,
+    "N": 6,
+    "O": 8,
+    "P": 2,
+    "Q": 1,
+    "R": 6,
+    "S": 4,
+    "T": 6,
+    "U": 4,
+    "V": 2,
+    "W": 2,
+    "X": 1,
+    "Y": 2,
+    "Z": 1,
+}
+    hand = []
+    full_letter_list = []
+    for alpha, number in LETTER_POOL.items():
+        for i in range(number):
+            full_letter_list.append(alpha)
+    while len(hand) != 10:
+        letter = random.choice(full_letter_list)
+        hand.append(letter)
+        full_letter_list.remove(letter)    
```

Comment:
Since the time complexity of .remove() on a list is O(n) and you have line 40 within a loop, while it doesn't look like it we've actually got a nested loop since under the hood Python has to loop through full_letter_list to find the letter to remove of it. So using remove() here make the time complexity of your solution quadratic. 

Consider using a different method (a different data structure) to keep track of the counts per letter that we can look up and modify in constant time. Maybe a dictionary?